pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -273,6 +277,37 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		Status: operatorv1.ConditionFalse,
 	}
 
+	if !preconditionsReady {
+		var message string
+		for _, err := range errs {
+			message = message + err.Error() + "\n"
+		}
+		if len(message) == 0 {
+			message = "the operator didn't specify what preconditions are missing"
+		}
+
+		// we are degraded, not available and we are not progressing
+
+		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
+		deploymentDegradedCondition.Reason = "PreconditionNotReady"
```

Comment:
ready is fixed term in kube. PreconditionNotFulfilled?

Reply:
done.